config/config_mgmt.py

```diff
@@ -0,0 +1,864 @@
+        try:
+            self.sysLog(msg="delPorts ports:{} force:{}".format(ports, force))
+
+            print('\nStart Port Deletion')
+            deps = list()
+
+            # Get all dependecies for ports
+            for port in ports:
+                xPathPort = self.sy.findXpathPortLeaf(port)
+                print('Find dependecies for port {}'.format(port))
+                self.sysLog(msg='Find dependecies for port {}'.format(port))
+                # print("Generated Xpath:" + xPathPort)
+                dep = self.sy.find_data_dependencies(str(xPathPort))
+                if dep:
+                    deps.extend(dep)
+
+            # No further action with no force and deps exist
+            if force == False and deps:
+                return configToLoad, deps, False;
+
+            # delets all deps, No topological sort is needed as of now, if deletion
+            # of deps fails, return immediately
+            elif deps and force:
+                for dep in deps:
+                    self.sysLog(msg='Deleting {}'.format(dep))
+                    self.sy.deleteNode(str(dep))
+            # mark deps as None now,
+            deps = None
+
+            # all deps are deleted now, delete all ports now
+            for port in ports:
+                xPathPort = self.sy.findXpathPort(port)
+                print("Deleting Port: " + port)
+                self.sysLog(msg='Deleting Port:{}'.format(port))
+                self.sy.deleteNode(str(xPathPort))
+
+            # Let`s Validate the tree now
+            if self.validateConfigData()==False:
+                return configToLoad, deps, False;
+
+            # All great if we are here, Lets get the diff
+            self.configdbJsonOut = self.sy.getData()
+            # Update configToLoad
+            configToLoad = self._updateDiffConfigDB()
+
+        except Exception as e:
+            print(e)
+            print("Port Deletion Failed")
+            return configToLoad, deps, False
+
+        return configToLoad, deps, True
+
+    def _addPorts(self, ports=list(), portJson=dict(), loadDefConfig=True):
+        '''
+        Add ports and default confug in data tree, validate and return resultant
+        config.
+
+        Parameters:
+            ports (list): list of ports
+            portJson (dict): Config DB json Part of all Ports, generated from
+                platform.json.
+            loadDefConfig: If loadDefConfig, add default config for ports as well.
+
+        Returns:
+            (configToLoad, ret) (tuple)[dict, bool]
+        '''
+        configToLoad = None
+        try:
+            self.sysLog(msg='Start Port Addition')
+            self.sysLog(msg="addPorts ports:{} loadDefConfig:{}".\
+                format(ports, loadDefConfig))
+            self.sysLog(msg="addPorts Args portjson {}".format(portJson))
+
+            print('\nStart Port Addition')
+
+            if loadDefConfig:
+                defConfig = self._getDefaultConfig(ports)
```

Comment:
This is the only place, where i/p arg ports is used. So if loadDefConfig is False, then this arg is not used.

Please help clarify

Reply:
yes, 
since python-click needs to calculate both ARGs, we pass both in here. This remains as is, as per own low level design doc. Till now for all breakout modes, I think portJson.keys() == ports.
So we can get rid of one, but let me ask once in the team, because we need to change caller code as well.